ocaml/xapi/updateinfo.mli

```diff
@@ -113,19 +117,30 @@ module Severity : sig
   val of_string : string -> t
 end
 
-(** The metadata of one update in updateinfo *)
+(** The type of [guidance] in updateinfo metadata. *)
+module GuidanceInUpdateInfo : sig
+  type t = (Guidance.kind * Guidance.t list) list
+
+  val default : t
+
+  val of_xml : Xml.xml list -> t
+
+  val to_json : t -> Yojson.Basic.t
+
+  val to_string : t -> string
+end
+
+(** The metadata of one update in updateinfo. *)
 module UpdateInfo : sig
   type t = {
       id: string
     ; summary: string
     ; description: string
-    ; rec_guidance: Guidance.t option
-    ; abs_guidance: Guidance.t option
+    ; guidance: GuidanceInUpdateInfo.t
```

Comment:
As we discussed, keep the name `guidance` align with the key in updateinfo.xml is ok, it is better to add a comment here that it is not one guidance for quick understanding.

Reply:
I think the type should be good to explain the structure.